File: Views/BaseView.vb
'
' DotNetNuke® - http://www.dotnetnuke.com
' Copyright (c) 2002-2011 by DotNetNuke Corp. 

'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'
Option Strict On

Imports System
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.HttpUtility
Imports System.Web.Security
Imports System.Text
Imports System.Configuration
Imports DotNetNuke
Imports DotNetNuke.Services.Localization
Imports DotNetNuke.Modules.Gallery.Config
Imports DotNetNuke.Modules.Gallery.Utils

Namespace DotNetNuke.Modules.Gallery.Views

  ''' <summary>
  ''' Base object used by various gallery view code
  ''' </summary>
  ''' <remarks></remarks>
  Public Class BaseView

    Private mGalleryControl As GalleryControl

    Public Sub New(ByVal GalleryControl As GalleryControl)
      mGalleryControl = GalleryControl
    End Sub 'New

    Public ReadOnly Property GalleryControl() As GalleryControl
      Get
        Return mGalleryControl
      End Get
    End Property

    Public ReadOnly Property Controls() As ControlCollection
      Get
        Return mGalleryControl.Controls
      End Get
    End Property

    Public Overridable Sub CreateChildControls()
    End Sub 'CreateChildControls

    Public Overridable Sub OnPreRender()
    End Sub 'OnPreRender

    Public Overridable Sub Render(ByVal wr As HtmlTextWriter)
    End Sub 'Render

    Protected Sub RenderTableBegin(ByVal wr As HtmlTextWriter, ByVal ClassName As String) ' Begin table in which we will render object (cellspacing, cellpadding, borderwidth)

      wr.AddAttribute(HtmlTextWriterAttribute.Class, ClassName)
      wr.AddAttribute(HtmlTextWriterAttribute.Id, "GalleryContent")
      wr.RenderBeginTag(HtmlTextWriterTag.Table)
    End Sub 'RenderTableBegin

    Protected Sub RenderTableEnd(ByVal wr As HtmlTextWriter)
      wr.RenderEndTag()
    End Sub ' End table in which object was rendered

    Protected Sub RenderInfo(ByVal wr As HtmlTextWriter, ByVal Info As String)
      wr.RenderBeginTag(HtmlTextWriterTag.Tr)

      wr.AddAttribute(HtmlTextWriterAttribute.Class, "Gallery_AlbumEmpty")
      wr.RenderBeginTag(HtmlTextWriterTag.Td)
      wr.Write(Info)
      wr.RenderEndTag() ' </td>

      wr.RenderEndTag() ' </tr>
    End Sub

    Public Shared Sub RenderImage(ByVal wr As HtmlTextWriter, ByVal ImageURL As String, ByVal Tooltip As String, ByVal Css As String)
      wr.AddAttribute(HtmlTextWriterAttribute.Border, "0")
      wr.AddAttribute(HtmlTextWriterAttribute.Src, ImageURL)

      If Css.Length > 0 Then
        wr.AddAttribute(HtmlTextWriterAttribute.Class, Css)
      End If

      'xhtml requires an alt tag, even if it is empty - HWZassenhaus 9/28/2008
      If Tooltip.Length > 0 Then
        wr.AddAttribute(HtmlTextWriterAttribute.Title, Tooltip)
        wr.AddAttribute(HtmlTextWriterAttribute.Alt, Tooltip)
      Else
        wr.AddAttribute(HtmlTextWriterAttribute.Alt, "")
      End If

      wr.RenderBeginTag(HtmlTextWriterTag.Img) '<img>
      wr.RenderEndTag() '/>'

    End Sub

    Public Shared Sub RenderImageButton(ByVal wr As HtmlTextWriter, ByVal URL As String, ByVal ImageURL As String, ByVal Tooltip As String, ByVal Css As String)
      wr.AddAttribute(HtmlTextWriterAttribute.Href, URL.Replace("~/", ""))
      wr.RenderBeginTag(HtmlTextWriterTag.A)
      RenderImage(wr, ImageURL, Tooltip, Css)
      wr.RenderEndTag() '</a>
    End Sub

    Public Shared Sub RenderCommandButton(ByVal wr As HtmlTextWriter, ByVal URL As String, ByVal Text As String, ByVal Css As String)
      wr.AddAttribute(HtmlTextWriterAttribute.Href, URL.Replace("~/", ""))

      If Css.Length > 0 Then
        wr.AddAttribute(HtmlTextWriterAttribute.Class, Css)
      End If

      wr.RenderBeginTag(HtmlTextWriterTag.A)
      wr.Write(Text)
      wr.RenderEndTag() '</a>
    End Sub

  End Class

End Namespace

File: Views/BaseView.cs
using DotNetNuke;
using DotNetNuke.Common;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Entities;
using DotNetNuke.Framework;
using DotNetNuke.Security;
using DotNetNuke.Services;
using DotNetNuke.Services.Exceptions;
using DotNetNuke.Services.Localization;
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Collections.Specialized;
using System.Configuration;
using System.Text;
using System.Text.RegularExpressions;
using System.Web;
using System.Web.Caching;
using System.Web.SessionState;
using System.Web.Security;
using System.Web.Profile;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
//
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2011 by DotNetNuke Corp. 

//
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//

using static System.Web.HttpUtility;
using static DotNetNuke.Modules.Gallery.Config;
using static DotNetNuke.Modules.Gallery.Utils;

namespace DotNetNuke.Modules.Gallery.Views
{

	/// <summary>
	/// Base object used by various gallery view code
	/// </summary>
	/// <remarks></remarks>
	public class BaseView
	{


		private GalleryControl mGalleryControl;
		public BaseView(GalleryControl GalleryControl)
		{
			mGalleryControl = GalleryControl;
		}
		//New

		public GalleryControl GalleryControl {
			get { return mGalleryControl; }
		}

		public ControlCollection Controls {
			get { return mGalleryControl.Controls; }
		}

		public virtual void CreateChildControls()
		{
		}
		//CreateChildControls

		public virtual void OnPreRender()
		{
		}
		//OnPreRender

		public virtual void Render(HtmlTextWriter wr)
		{
		}
		//Render

		// Begin table in which we will render object (cellspacing, cellpadding, borderwidth)
		protected void RenderTableBegin(HtmlTextWriter wr, string ClassName)
		{

			wr.AddAttribute(HtmlTextWriterAttribute.Class, ClassName);
			wr.AddAttribute(HtmlTextWriterAttribute.Id, "GalleryContent");
			wr.RenderBeginTag(HtmlTextWriterTag.Table);
		}
		//RenderTableBegin

		protected void RenderTableEnd(HtmlTextWriter wr)
		{
			wr.RenderEndTag();
		}
		// End table in which object was rendered

		protected void RenderInfo(HtmlTextWriter wr, string Info)
		{
			wr.RenderBeginTag(HtmlTextWriterTag.Tr);

			wr.AddAttribute(HtmlTextWriterAttribute.Class, "Gallery_AlbumEmpty");
			wr.RenderBeginTag(HtmlTextWriterTag.Td);
			wr.Write(Info);
			wr.RenderEndTag();
			// </td>

			wr.RenderEndTag();
			// </tr>
		}

		public static void RenderImage(HtmlTextWriter wr, string ImageURL, string Tooltip, string Css)
		{
			wr.AddAttribute(HtmlTextWriterAttribute.Border, "0");
			wr.AddAttribute(HtmlTextWriterAttribute.Src, ImageURL);

			if (Css.Length > 0) {
				wr.AddAttribute(HtmlTextWriterAttribute.Class, Css);
			}

			//xhtml requires an alt tag, even if it is empty - HWZassenhaus 9/28/2008
			if (Tooltip.Length > 0) {
				wr.AddAttribute(HtmlTextWriterAttribute.Title, Tooltip);
				wr.AddAttribute(HtmlTextWriterAttribute.Alt, Tooltip);
			} else {
				wr.AddAttribute(HtmlTextWriterAttribute.Alt, "");
			}

			wr.RenderBeginTag(HtmlTextWriterTag.Img);
			//<img>
			wr.RenderEndTag();
			///>'

		}

		public static void RenderImageButton(HtmlTextWriter wr, string URL, string ImageURL, string Tooltip, string Css)
		{
			wr.AddAttribute(HtmlTextWriterAttribute.Href, URL.Replace("~/", ""));
			wr.RenderBeginTag(HtmlTextWriterTag.A);
			RenderImage(wr, ImageURL, Tooltip, Css);
			wr.RenderEndTag();
			//</a>
		}

		public static void RenderCommandButton(HtmlTextWriter wr, string URL, string Text, string Css)
		{
			wr.AddAttribute(HtmlTextWriterAttribute.Href, URL.Replace("~/", ""));

			if (Css.Length > 0) {
				wr.AddAttribute(HtmlTextWriterAttribute.Class, Css);
			}

			wr.RenderBeginTag(HtmlTextWriterTag.A);
			wr.Write(Text);
			wr.RenderEndTag();
			//</a>
		}

	}

}
